lib/daru/dataframe.rb

```diff
@@ -906,19 +906,19 @@ def filter_vector vec
     def filter_rows &block
       return to_enum(:filter_rows) unless block_given?
 
-      df = Daru::DataFrame.new({}, order: @vectors.to_a)
-      marked = []
-
-      @index.each do |index|
-        keep_row = yield access_row(index)
-        marked << index if keep_row
-      end
-
-      marked.each do |idx|
-        df.row[idx] = self[idx, :row]
+      values = to_a[0].map{|h| h.values}
```

Comment:
Could you try `self.to_matrix.to_a` to get rows in an Array of Arrays and see if it's faster? I have a hunch it might be.